packages/@uppy/companion/src/server/controllers/send-token.js

```diff
@@ -17,7 +18,7 @@ const htmlContent = (token, origin) => {
     <head>
         <meta charset="utf-8" />
         <script>
-          window.opener.postMessage(${serialize({ token })}, ${serialize(origin)})
+          if (window.opener) window.opener.postMessage(${serialize({ token })}, ${serialize(origin)})
```

Comment:
given that window.opener is null, next line is not executed when `window.close`.

Reply:
According to <https://developer.mozilla.org/en-US/docs/Web/API/Window/opener>:

> In the following cases, the browser does not populate window.opener, but leaves it null:
> 
> - The opener can be omitted by specifying rel=noopener on a link, or passing noopener in the windowFeatures parameter.
> - Windows opened because of links with a target of _blank don't get an opener, unless explicitly requested with rel=opener.
> - Having a Cross-Origin-Opener-Policy header with a value of same-origin prevents setting opener. Since the new window is loaded in a different browsing context, it won't have a reference to the opening window.

Reply:
I vaguely remember the discussion around the reason for not checking window.opener before calling it. IIRC the reason was that it is better that the page crashes with an error message in the browser console than having it just close the page without any error or anything happening. I think if we add this check we will not why it's not working if this happens again in the future.